test/net-spv-test.js

```diff
@@ -211,16 +214,56 @@ describe('SPV', function() {
 
       // Restore
       const waiter2 = new Promise((res, rej) => {
+        let completed = false;
         spv.on('connect', (entry) => {
-          if (entry.height === full.chain.height)
+          if (!completed && entry.height === full.chain.height) {
+            completed = true;
             res();
+          }
         });
       });
       await spv.chain.removeInvalid(entry.hash);
       await waiter2;
       assert.strictEqual(full.chain.height, spv.chain.height);
     });
 
+    it('should resync multiple times', async () => {
+      const resetHeight = 1;
+      const fullNodeHeight = full.chain.height;
+
+      // Returns a promise that waits for spv wdb to catch up to full node
+      // `completed` ensures that the promise will not resolve multiple times
+      const waiter = () => new Promise((resolve) => {
+        let completed = false;
+        spv.on('connect', (entry) => {
+          if (!completed && entry.height === fullNodeHeight) {
+            completed = true;
+            resolve();
+          }
+        });
+      });
+
+      // If rescans don't complete in time, throw error
+      const timer = setTimeout(() => {
+        throw new Error('SPV rescan stuck.');
+      }, 1000);
+
+      // Rescan once
+      await spv.chain.reset(resetHeight);
+      assert(spv.chain.height === resetHeight);
+      await waiter();
+      assert(spv.chain.height === fullNodeHeight);
+
+      // Rescan again
+      await spv.chain.reset(resetHeight);
+      assert(spv.chain.height === resetHeight);
+      await waiter();
```

Comment:
Oh I see the problem, it's because you are waiting for the same promise to resolve, twice.